app/src/main/java/com/x8bit/bitwarden/data/platform/manager/restriction/RestrictionManagerImpl.kt

```diff
@@ -0,0 +1,137 @@
+package com.x8bit.bitwarden.data.platform.manager.restriction
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.RestrictionsManager
+import android.os.Bundle
+import com.x8bit.bitwarden.data.platform.manager.AppForegroundManager
+import com.x8bit.bitwarden.data.platform.manager.dispatcher.DispatcherManager
+import com.x8bit.bitwarden.data.platform.manager.model.AppForegroundState
+import com.x8bit.bitwarden.data.platform.repository.EnvironmentRepository
+import com.x8bit.bitwarden.data.platform.repository.model.Environment
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+/**
+ * The default implementation of the [RestrictionManager].
+ */
+class RestrictionManagerImpl(
+    appForegroundManager: AppForegroundManager,
+    dispatcherManager: DispatcherManager,
+    private val context: Context,
+    private val environmentRepository: EnvironmentRepository,
+    private val restrictionsManager: RestrictionsManager,
+) : RestrictionManager {
+    private val ioScope = CoroutineScope(dispatcherManager.io)
+    private val intentFilter = IntentFilter(Intent.ACTION_APPLICATION_RESTRICTIONS_CHANGED)
+    private val restrictionsChangedReceiver = RestrictionsChangedReceiver()
+    private var isReceiverRegistered = false
+
+    init {
+        appForegroundManager
+            .appForegroundStateFlow
+            .onEach {
+                when (it) {
+                    AppForegroundState.BACKGROUNDED -> handleBackground()
+                    AppForegroundState.FOREGROUNDED -> handleForeground()
+                }
+            }
+            .launchIn(ioScope)
+    }
+
+    override fun updatePreconfiguredRestrictionSettings() {
+        restrictionsManager
+            .applicationRestrictions
+            ?.takeUnless { it.isEmpty }
+            ?.let { setPreconfiguredSettings(it) }
+    }
+
+    private fun handleBackground() {
+        if (isReceiverRegistered) {
+            context.unregisterReceiver(restrictionsChangedReceiver)
+        }
+        isReceiverRegistered = false
+    }
+
+    private fun handleForeground() {
+        context.registerReceiver(restrictionsChangedReceiver, intentFilter)
+        isReceiverRegistered = true
+        updatePreconfiguredRestrictionSettings()
+    }
+
+    private fun setPreconfiguredSettings(bundle: Bundle) {
+        bundle.keySet().forEach { key ->
```

Comment:
Thoughts on whether we actually need this loop?

We can easily just do this:
```kotlin
bundle
    .getString(BASE_ENVIRONMENT_URL_RESTRICTION_KEY)
    ?.let { url -> setPreconfiguredUrl(baseEnvironmentUrl = url) }
```

And just keep adding more in the future if needed.

Reply:
I agree, the loop seems unnecessary to me.